src/java_bytecode/expr2java.cpp

```diff
@@ -235,6 +237,63 @@ std::string expr2javat::convert_constant(
     dest+='L';
     return dest;
   }
+  else if((src.type()==java_float_type()) ||
+          (src.type()==java_double_type()))
+  {
+    ieee_floatt ieee_repr(to_constant_expr(src));
+    std::string result;
+
+    bool float_flag=(src.type()==java_float_type());
+    if(ieee_repr.is_zero())
+    {
+      result+='0';
+      format_spect fs;
+      if(fs.precision>0)
```

Comment:
Thinking further, I suspect we want 0.0 at all times here? We want an expression that is unambiguously zero of the given type; I don't see that 0.000000 (by default init of format_spect) is any better.